src/main/scala/nl.knaw.dans.easy.deposit/servlets/package.scala

```diff
@@ -61,30 +61,36 @@ package object servlets extends DebugEnhancedLogging {
   val contentTypeJson: (String, String) = "content-type" -> "application/json;charset=UTF-8"
   val contentTypePlainText: (String, String) = "content-type" -> "text/plain;charset=UTF-8"
 
-  implicit class RichManagedArchiveInputStream(val archiveInputStream: ManagedResource[ArchiveInputStream]) extends AnyVal {
-    def unpackPlainEntriesTo(dir: File, id: => UUID): Try[Unit] = {
-      archiveInputStream.apply(_.unpackPlainEntriesTo(dir, id))
-    }
-  }
-
   implicit class RichArchiveInputStream(val archiveInputStream: ArchiveInputStream) extends AnyVal {
 
-    def unpackPlainEntriesTo(targetDir: File, id: => UUID): Try[Unit] = {
-      def extract(entry: ArchiveEntry): Try[Unit] = {
-        if (!(targetDir / entry.getName).isChildOf(targetDir))
-          Failure(MalformedArchiveException(s"Can't extract ${ entry.getName }"))
+    /**
+     *
+     * @param targetDir    temporary file to stage uploads
+     * @param draftDeposit deposit receiving the uploads (info for error message)
+     * @param path         relative path for the upload root (info for error message)
+     * @param uploadName   of multipart item of request (client point of view, info for error message)
+     * @return
+     */
+    def unpackPlainEntriesTo(targetDir: File, draftDeposit: => DepositDir, path: => Path, uploadName: String): Try[Unit] = {
+      def extractOne(entry: ArchiveEntry): Try[Unit] = {
+        lazy val clientMsg = s"Can't extract ${ entry.getName } from $uploadName"
+        val targetFile = targetDir / entry.getName
+        if (!targetFile.isChildOf(targetDir)) {
+          Failure(MalformedArchiveException(clientMsg + ", invalid path"))
+        }
         else if (entry.isDirectory) {
-          Try((targetDir / entry.getName).createDirectories())
+          Try(targetFile.createDirectories())
         }
         else {
-          logger.debug(s"[$id] Extracting ${ entry.getName }")
+          lazy val logMsg = s"[${ draftDeposit.id }] Extracting ${ entry.getName } from $uploadName"
+          logger.debug(logMsg)
```

Comment:
I thought you mentioned previously how `logger.debug` evaluates its arguments lazily... Why do you then extract the `logMsg` as a `lazy val`?